signal/diagnostics.go

```diff
@@ -0,0 +1,122 @@
+package signal
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/dcos/dcos-signal/config"
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/segmentio/analytics-go"
+)
+
+// HealthReport defines the JSON received from the /system/health/report endpoint
+// The health report returns keys that are not formatted for JSON specifically, so
+// we do not modify them and instead pass the param as the key, unmodified.
+type HealthReport struct {
```

Comment:
@darkonie Can you verify any changes to 3DT which might break this struct? Specifically Unit and Node types?